### PR TITLE
Add support for disable default snat

### PR DESCRIPTION
**Pull request by okguru1**
Adding support for `disable_default_snat`, which (as the name suggests) disables the default snat. This option is available on the root resource, so all this PR does is expose this option.

---
**Checklist**

I applicable, I acknowledge that I have:
- [X] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [X] Ran `terraform fmt` on all modified files
- [X] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [X] Made sure all relevant tests pass

### CODE REVIEW

**Review by ludoo — COMMENTED** [review]
Thanks for this. A couple comments below.

Main one is we work very hard to minimize the variable surface, and I would prefer not adding one variable for a tiny feature. This should be an additional optional attribute in one of the existing `_config` variables, possibly `vpc_config` as it's the one most closely related to this.

The second comment is I would prefer using a dynamic block and a `null` default, so that the block is transparently skipped when the attribute is not explicitly set. This prevented issues with recreation of existing clusters for other attributes, and we might want to adopt the same approache here, even just for consistency's sake.

And last, do you know if this is also supported for autopilot? If it is, we should update both cluster modules together.

**Comment by okguru1** [comment]
@ludoo thanks for the prompt feedback. adjusted the PR to utilize a dynamic block (with conditional on null), had the variable moved inside `vpc_config` (with default null), and aligned the autopilot module the same way.

**Comment by ludoo** [comment]
> @ludoo thanks for the prompt feedback. adjusted the PR to utilize a dynamic block (with conditional on null), had the variable moved inside `vpc_config` (with default null), and aligned the autopilot module the same way.

super quick :)

**Comment by ludoo** [comment]
Can you pull and merge master? I cannot use your fork...

File: modules/gke-cluster-autopilot/main.tf [hunk 1]
```diff
@@ -94,6 +94,13 @@ resource "google_container_cluster" "cluster" {
     }
   }
 
+  dynamic "default_snat_status" {
+    for_each = var.vpc_config.disable_default_snat == null ? [""] : []
```
Comment (ludoo) [inline modules/gke-cluster-autopilot/main.tf:98]: I would use something like

```hcl
dynamic "default_snat_status" {
  for_each = var.vpc_config.disable_default_snat == null ? [] : [""]
  content {
    disabled = var.vpc_config.disable_default_snat
  }
}
```

Comment (okguru1) [reply modules/gke-cluster-autopilot/main.tf:98]: good catch 😅

**Comment by okguru1** [comment]
> Can you pull and merge master? I cannot use your fork...

yes, updated my master branch, or did you mean something else?

**Review by ludoo — APPROVED** [review]
last fix on redundant default optional value then this is good to go, many thanks :)

File: modules/gke-cluster-standard/variables.tf [hunk 2]
```diff
@@ -421,6 +422,7 @@ variable "release_channel" {
 variable "vpc_config" {
   description = "VPC-level configuration."
   type = object({
+    disable_default_snat       = optional(bool, null)
```
[the remark this answers is not in the document]
Comment (okguru1) [reply modules/gke-cluster-standard/variables.tf]: done

**Comment by ludoo** [comment]
Thanks a lot for your patience :)

**Comment by ludoo** [comment]
ach, tfdoc strikes again :)

**Comment by okguru1** [comment]
> ach, tfdoc strikes again :)

hahaha yea missed this, if I continue contributing I should probably make a precommit with all of these :)

**Comment by ludoo** [comment]
> > ach, tfdoc strikes again :)
> 
> hahaha yea missed this, if I continue contributing I should probably make a precommit with all of these :)

hope you do :)

**Comment by ludoo** [comment]
Oh God, one more merge :) if you set your fork to allow changes from this repo I could push the button for you

**Comment by okguru1** [comment]
> Oh God, one more merge :) if you set your fork to allow changes from this repo I could push the button for you

done, I have to learn how to clickops those fork permissions, will do for next times

**Pull request merged**